src/IDataProvider.ts

```diff
@@ -1,4 +1,5 @@
 import { IOracleData, IFlatObj, IDataType } from "./IOracleData"
 
+export type IDataProviderRequestArg = { name: string, type: IDataType }
 export type IDataProvider<TConfig extends IFlatObj, TArgs extends any[]> = (config: TConfig, ...args: TArgs) => Promise<IOracleData>
-export type ITypeProvider<TConfig extends IFlatObj> = (config: TConfig) => { type: IDataType, name: string }
\ No newline at end of file
+export type ITypeProvider<TConfig extends IFlatObj> = (config: TConfig) => { type: IDataType, name: string, args: IDataProviderRequestArg[] }
```

Comment:
May we fix max-line-size ? in order to avoid long one-liners (i.e. nested structures).